### PR TITLE
[SPARK-9023] [SQL] Followup for #7456 (Efficiency improvements for UnsafeRows in Exchange)

### PR DESCRIPTION
**Pull request by JoshRosen**
This patch addresses code review feedback from #7456.

### CODE REVIEW

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java [hunk 1]
```diff
@@ -365,6 +368,36 @@ public InternalRow copy() {
...
+   *
+   * @param out the stream to write to.
+   * @param writeBuffer a byte array for buffering chunks of off-heap data while writing to the
+   *                    output stream. If this row is backed by an on-heap byte array, then this
+   *                    buffer will not be used and may be null.
+   */
+  public void writeToStream(OutputStream out, byte[] writeBuffer) throws IOException {
+    if (baseObject instanceof byte[]) {
+      int offsetInByteArray = (int) (PlatformDependent.BYTE_ARRAY_OFFSET - baseOffset);
+      out.write((byte[]) baseObject, offsetInByteArray, sizeInBytes);
```
Comment (rxin) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java]: does this write always write everything?

**Comment by rxin** [comment]
LGTM - just that one thing. Can you verify? Feel free to merge once you verify it writes the full byte array out always.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java (hunk 1 above)
Comment (JoshRosen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java]: Yes; `OutputStream.write()` is a `void` method.  It's `read` that you have to be careful with.

**Comment by rxin** [comment]
I merged it.

**Pull request closed without merging**